### PR TITLE
JS Task 2

### PR DESCRIPTION
**Pull request by KamilWasilonek**
https://kamilwasilonek.github.io/ui-internship-lwo2/html/Axit/

### CODE REVIEW

File: html/Axit/js/tabs/helper-functions.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
+function highlightButton(btn) {
+  btn.classList.add('tabs__btn--active');
+}
+
+function removeButtonActiveClass(buttons) {
+  buttons.forEach((element) => {
```
Comment (tkachukivan) [inline html/Axit/js/tabs/helper-functions.js:6]: what if you have 10/20 tabs or even more, but every time have to iterate throw all of them

File: html/Axit/js/tabs/helper-functions.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+function hideContent(contentList) {
+  for (let i = 0; i < contentList.length; i++) {
+    // eslint-disable-next-line no-param-reassign
+    contentList[i].style.opacity = '0';
+  }
+}
+
+function showElement(elementToShow) {
+  // eslint-disable-next-line no-param-reassign
+  elementToShow.style.opacity = '1';
```
Comment (tkachukivan) [inline html/Axit/js/tabs/helper-functions.js:20]: classes? :)

File: html/Axit/js/tabs/helper-functions.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  for (let i = 0; i < contentList.length; i++) {
+    if (id === contentList[i].dataset.id) {
+      return contentList[i];
+    }
+  }
+}
+
+function changeSectionHeight(visibleTab, tabs) {
+  if (window.innerWidth <= 767) {
+    let newHeight = visibleTab.offsetHeight + 240;
```
Comment (tkachukivan) [inline html/Axit/js/tabs/helper-functions.js:33]: what is a reason for changing height of element in tabs? isn't changing height of element by content default behaviour?

File: html/Axit/js/tabs/tab-switching.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  findContent,
+  changeSectionHeight,
+} from './helper-functions.js';
+
+export function addTabSwitchingEffect() {
+  components.then(
+      ({tabsButtonsList, contentList, activeTab, tabs, buttons}) => {
+        (function initFirstTab() {
+        // eslint-disable-next-line no-param-reassign
+          activeTab.style.opacity = 1;
```
Comment (tkachukivan) [inline html/Axit/js/tabs/tab-switching.js:16]: no real reason for this, as user will not even see this animation as tabs not in first screen

File: html/Axit/js/tabs/tab-switching.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+        changeSectionHeight(activeTab, tabs);
+
+        window.addEventListener('resize', () => {
+          changeSectionHeight(activeTab, tabs);
+        });
+
+        tabsButtonsList.addEventListener('click', (event) => {
+          let target = event.target;
+          if (target.tagName === 'BUTTON') {
```
Comment (tkachukivan) [inline html/Axit/js/tabs/tab-switching.js:27]: usually it loos a bit more readable when in if call return, at place and write other code in same level
```
if (target.tagName !== 'BUTTON') { return }
//rest code here
```

**Review by tkachukivan — COMMENTED** [review]
I would start animation of appearing new element after old one hidden, it would look a bit prettier 

responsive version looks great :)

File: html/Axit/js/tabs/tab-switching.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+  hideContent,
+  showElement,
+  findContent,
+  changeSectionHeight,
+} from './helper-functions.js';
+
+export function addTabSwitchingEffect() {
+  components.then(
+      ({tabsButtonsList, contentList, activeTab, tabs, buttons}) => {
+        (function initFirstTab() {
```
Comment (tsvetkovv) [inline html/Axit/js/tabs/tab-switching.js:14]: why self-invoking function?

File: html/Axit/js/tabs/helper-functions.js (hunk 2 above)
Comment (tsvetkovv) [reply html/Axit/js/tabs/helper-functions.js:20]: ..for parent component to hide every content item to avoid traversing

**Pull request merged**